src/rgw/rgw_rados.cc

```diff
@@ -8097,6 +8107,13 @@ int RGWRados::copy_obj(RGWObjectCtx& obj_ctx,
   if (ret < 0) {
     return ret;
   }
+  if (rgw_s3_is_encrypted(src_attrs)) {
+    // Current encryption implemenation rely on the part sequence in manifest
+    // but current copy operation will result in part sequence change
+    // To Do: need more comprehensive design to support this kind of copy operation
+    ldout(cct, 0) << "ERROR: failed to copy encrypted multipart object " << src_obj << dendl;
```

Comment:
the log message and comment still refer to multipart objects, but this code path applies to non-multipart objects too